data/src/history/manager.rs

```diff
@@ -491,6 +498,43 @@ impl Data {
 
                     true
                 }
+                Source::Internal(source::Internal::Status(status)) => {
+                    let source: source::Server;
+                    match status {
+                        source::Status::Success => {
+                            source = source::Server::new(source::server::Kind::StatusSuccess, None);
+                        }
+                        source::Status::Error => {
+                            source = source::Server::new(source::server::Kind::StatusError, None);
+                        }
+                    };
```

Comment:
Once the other comment is addressed, this is unnecessary since we should be able to use `Source::Internal(internal)` value to figure out if the config option is set

Reply:
You were right! Got rid of this section in the refactor.